snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedMemberOfFieldFixRequest.java

```diff
@@ -0,0 +1,252 @@
+						missingActiveMembersOf.put(m.getReferencedComponentId(), referenceSetId);
+					}
+					missingMembersOf.put(m.getReferencedComponentId(), referenceSetId);
+				})
+			);
+			
+			SearchResourceRequestIterator<SnomedConceptSearchRequestBuilder, SnomedConcepts> memberOfonceptIterator = 
+					new SearchResourceRequestIterator<>(
+							SnomedRequests.prepareSearchConcept().isMemberOf(referenceSetId).setFields("id"),
+							b -> b.setLimit(50_000).build().execute(context));
+			memberOfonceptIterator.forEachRemaining(concepts -> concepts.forEach(concept -> missingMembersOf.removeAll(concept.getId())));
+			missingMembersOfConcepts.putAll(missingMembersOf);
+			log.info("Found " + missingMembersOf.size() + " concepts with missing member of entry for reference set " + referenceSetId);
+			
+			SearchResourceRequestIterator<SnomedConceptSearchRequestBuilder, SnomedConcepts> activeMemberOfonceptIterator = 
+					new SearchResourceRequestIterator<>(
+							SnomedRequests.prepareSearchConcept().isActiveMemberOf(referenceSetId).setFields("id"),
+							b -> b.setLimit(50_000).build().execute(context));
+			activeMemberOfonceptIterator.forEachRemaining(concepts -> concepts.forEach(concept -> missingActiveMembersOf.removeAll(concept.getId())));
+			missingActiveMembersOfConcepts.putAll(missingActiveMembersOf);
+			log.info("Found " + missingActiveMembersOf.size() + " concepts with missing active member of entry for reference set " + referenceSetId);
+		}
+		
+		int modifiedConceptCount = 0;
+		
+		Set<String> allAffectedConcepts = Sets.union(missingMembersOfConcepts.keySet(), missingActiveMembersOfConcepts.keySet());
+		
+		for (String conceptId : allAffectedConcepts) {
+			try {
+				
+				final SnomedConceptDocument concept = context.lookup(conceptId, SnomedConceptDocument.class);				
+				List<String> memberOf = new ArrayList<>(concept.getMemberOf());
+				if (missingMembersOfConcepts.containsKey(conceptId)) {
+					memberOf.addAll(missingMembersOfConcepts.get(conceptId));				
+				}
+				
+				List<String> activeMemberOf = new ArrayList<>(concept.getActiveMemberOf());
+				if (missingActiveMembersOfConcepts.containsKey(conceptId)) {
+					activeMemberOf.addAll(missingActiveMembersOfConcepts.get(conceptId));
+				}
+				
+				SnomedConceptDocument updatedDocument = SnomedConceptDocument.builder(concept)
+						.memberOf(memberOf)
+						.activeMemberOf(activeMemberOf)
+						.build();
+				context.update(concept, updatedDocument);
+				
+				modifiedConceptCount++;
+				if (modifiedConceptCount%LIMIT == 0) {
+					context.commit("Update memberOf/activeMemberOf fields on concepts");
+				}
+				
+			} catch (ComponentNotFoundException exception) {
+				log.error(exception.getMessage());
+			}
+		}
+		
+		//Commit any remaining concepts;
+		context.commit("Update memberOf/activeMemberOf fields on concepts");
+		affectedComponentIds.addAll(allAffectedConcepts);
+	}
+	
+	private void processDescriptions(TransactionContext context, Logger log) {		
```

Comment:
Would be nice to merge these two methods together, unless it is a too big effort given the short timeframe.